### PR TITLE
CLOUDP-47519: Tidy up README

### PR DESCRIPTION
**Pull request by fabianlindfors**
As we are preparing for release we want a friendly and welcoming README. I've tried to give a very brief introduction to what the broker does and list the things it can do.

I also moved all development specific info to the `docs` folder. This is the folder we are using for CLOUDP-46569 (#13). The "Testing in Kubernetes" section can probably be removed as part of that PR as they document the same process, but #13 is much more detailed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,14 @@
 # MongoDB Atlas Service Broker
 
-Implementation of the [Open Service Broker API](https://www.openservicebrokerapi.org/) for [MongoDB Atlas](https://www.mongodb.com/cloud/atlas). Deploy this service to easily manage Atlas instances!
+Use the Atlas Service Broker to connect [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) to any platform which supports the [Open Service Broker API](https://www.openservicebrokerapi.org/), such as [Kubernetes](https://kubernetes.io/) and [Pivotal Cloud Foundry](https://pivotal.io/platform).
```
Comment (jamesbroadhead) [inline README.md]: "connect to"  ... "from any platform" ...

File: README.md [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 
+- Provision managed MongoDB clusters on Atlas directly from your platform of choice. Includes support for all cluster configuration settings and cloud providers available on Atlas.
+- Manage and scale clusters without leaving your platform.
+- Create bindings to allow your applications access to clusters.
+
+## Documentation
+
+For instructions on how to install and use the MongoDB Atlas Service Broker please refer to the [documentation](https://docs.mongodb.com/atlas-open-service-broker).
```
Comment (jamesbroadhead) [inline README.md]: 404 atm - is this definitely where they'll be?

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+   - `public-key`for the API key
+   - `private-key`for the API key
+5. Deploy the service broker by running `scripts/kubernetes-deploy.sh <namespace>`. This will create
+   a new deployment and a service of the image from step 2. The script will also deploy the actual service broker resource with the
+   name `atlas-service-broker`.
+6. Make sure the broker is ready by running `svcat get brokers`.
+7. A new instance can be provisioned by running `kubectl create -f
+   scripts/kubernetes/instance.yaml`. The instance will be given the name `atlas-cluster-instance`
+   and its status can be checked using `svcat get instances`.
+8. Once the instance is up and running a binding can be created to gain access. A binding named
```
Comment (BenElgar) [inline docs/README.md]:
```suggestion
8. Once the instance is up and running, a binding can be created to gain access. A binding named
```

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+   `atlas-cluster-binding` can be created by running `kubectl create -f
+   script/kubernetes/binding.yaml`. The binding credentials will automatically be stored in a secret
+   of the same name.
+9. After use all bindings can be removed by running `svcat unbind atlas-cluser-instance` and the
```
Comment (BenElgar) [inline docs/README.md]:
```suggestion
9. After use, all bindings can be removed by running `svcat unbind atlas-cluser-instance` and the
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,14 @@
 # MongoDB Atlas Service Broker
 
-Implementation of the [Open Service Broker API](https://www.openservicebrokerapi.org/) for [MongoDB Atlas](https://www.mongodb.com/cloud/atlas). Deploy this service to easily manage Atlas instances!
+Use the Atlas Service Broker to connect [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) to any platform which supports the [Open Service Broker API](https://www.openservicebrokerapi.org/), such as [Kubernetes](https://kubernetes.io/) and [Pivotal Cloud Foundry](https://pivotal.io/platform).
 
+- Provision managed MongoDB clusters on Atlas directly from your platform of choice. Includes support for all cluster configuration settings and cloud providers available on Atlas.
+- Manage and scale clusters without leaving your platform.
```
[the remark this answers is not in the document]
Comment (jamesbroadhead) [reply README.md]: It's clearer as-is imo

File: README.md [hunk 6]
```diff
@@ -20,78 +27,23 @@ Configuration is handled with environment variables.
...
-[Configuration](#configuration). These tests can be run with `go test -timeout 1h ./test`. The
-default timeout is 10 minutes which is normally too short for some of the tests, hence it's
-recommended to raise the timeout to 1 hour. As part of the integration tests a MongoDB connection is
-set up to test the generated credentials. For this test to not fail the testing host needs to be
-whitelisted in Atlas.
+### Logs
 
-Unit and integration tests can be run at once using `go test -timeout 1h ./...`. Remember
-to pass the necessary environment variables and raise the timeout limit.
+Logs are written to stderr and each line is in a structured JSON format.
```
[the remark this answers is not in the document]
Comment (jamesbroadhead) [reply README.md]: +1 -- if it's really stderr, we should change it to stdout

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenElgar) [reply README.md]: Fair enough—I don't feel particularly strongly.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (fabianlindfors) [reply README.md]: That's the default in `zap`, from what I can tell the operator uses the same. My understanding was that stderr should be used for diagnostics, wouldn't that include log messages? The broker isn't meant to be interacted with directly so it doesn't have any "real" output.

Example of other peoples' take on this:
https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabianlindfors) [reply README.md]: I thought of that but couldn't find any good page for Cloud Foundry and opted for consistency. I found one now though: https://pivotal.io/open-service-broker so I will change!

[the remark this answers is not in the document]
Comment (fabianlindfors) [reply README.md]: Or maybe not. This page is kind of generic: https://pivotal.io/open-service-broker. What do you think?

File: README.md (hunk 2 above)
Comment (fabianlindfors) [reply README.md]: According to Melissa, yes. Should we remove this section and add it in again once the docs are published?

**Comment by fabianlindfors** [comment]
I agree! `docs` is ambiguous at best.  How about a `dev` folder with this `README` at the root and then a `dev/scripts` folder?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenElgar) [reply README.md]: Not a big deal; whichever you'd prefer 🙂

File: README.md (hunk 2 above)
Comment (jamesbroadhead) [reply README.md]: nah, it's fine

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jamesbroadhead) [reply README.md]: I'm pretty sure that that means "diagnostic" as in "to diagnose a problem". 

If the README is currently factual, let's leave as-is. 

We should revisit stdout/stderr though -- could you file a ticket

[the remark this answers is not in the document]
Comment (fabianlindfors) [reply README.md]: Done: https://jira.mongodb.org/browse/CLOUDP-47817

**Pull request merged**